File: Roadmap/09 - HERENCIA/java/pguillo02.java
class Animal{
    final String name;
    final int peso;
    final int tamaño;

    public Animal(String name, int peso, int tamaño){
        this.name = name;
        this.tamaño = tamaño;
        this.peso = peso;
    }

    public void sound(){}
}

class Gato extends Animal{
    private final String sonido;

    public Gato(String name, int peso, int tamaño, String sonido){
        super(name, peso, tamaño);
        this.sonido = sonido;
    }

    @Override
    public void sound(){
        System.out.println("El" + name + "suena" + sonido);
    }
}

public class pguillo02 {

    public static void main(String args[]){
        Gato g = new Gato("Gato", 22, 22, "Miau");
        g.sound();
    }
}

File: Roadmap/09 - HERENCIA/python/pguillo02.py
class Animal:
    
    def __init__ (self, name, tamaño, peso):
        self.name = name
        self.tamaño = tamaño 
        self.peso = peso

    def sound(self):
        pass

class Gato(Animal):

    def __init__ (self, name, tamaño, peso, sonido):
        super().__init__(name, tamaño, peso)
        self.sonido = sonido

    def sound(self):
        print(f'El {self.name} suena {self.sonido}')


g = Gato("Gato", 22, 22, "Miau")
g.sound()
